### PR TITLE
[Documentation] `NeighborLoader` Tutorial

### PR DESCRIPTION
**Pull request by LukeLIN-web**
https://github.com/pyg-team/pytorch_geometric/issues/7892  
Neighbor Sampling: This tutorial should go more in-depth into our [NeighborLoader](https://pytorch-geometric.readthedocs.io/en/latest/modules/loader.html#torch_geometric.loader.NeighborLoader), explain its usage and reference corresponding examples. It should outline the general computation flow of GNNs with neighborhood sampling, and things to look out for (e.g., ensuring to only make use of the first batch_size many nodes for loss/metric computation). It should also cross-link to our ["Hierarchical Neighborhood Sampling"](https://pytorch-geometric.readthedocs.io/en/latest/advanced/hgam.html) tutorial as a simple extension to improve its efficiency.

### CODE REVIEW

**Comment by rusty1s** [comment]
Thanks! Feel free to ping me if you wanna discuss anything :)

**Comment by LukeLIN-web** [comment]
> 

I wonder whether it meets the requirements.

**Comment by LukeLIN-web** [comment]
> Thanks! Feel free to ping me if you wanna discuss anything :)

I wonder whether it meets the requirements now.

File: docs/source/tutorial/neighborloader.rst [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            for i, conv in enumerate(self.convs):
+                xs = []
+                for batch in subgraph_loader:
+                    x = x_all[batch.n_id.to(x_all.device)].to(device)
+                    x = conv(x, batch.edge_index.to(device))
+                    if i < len(self.convs) - 1:
+                        x = x.relu_()
+                    xs.append(x[:batch.batch_size].cpu()) # we only need the representations of the target nodes
+                x_all = torch.cat(xs, dim=0)
+            return x_all
```
Comment (rusty1s) [inline docs/source/tutorial/neighborloader.rst:124]: We would need to go a bit more into depth here if we wanna take about layer-wise computation of node embeddings for inference.

File: docs/source/tutorial/neighborloader.rst [hunk 2]
```diff
@@ -0,0 +1,169 @@
+NeighborLoader
+=====================================
+``NeighborLoader`` is a data loader that performs neighbor sampling as introduced in the “Inductive Representation Learning on Large Graphs” paper.
+This loader allows for mini-batch training of GNNs on large-scale graphs where full-batch training leads to out of memory problem.
```
Comment (rusty1s) [inline docs/source/tutorial/neighborloader.rst:4]: Can we provide a bit more detail about
* why we need sampling in general
* how it works on paper

**Review by rusty1s — COMMENTED** [review]
Thanks. I think this looks great. Left two comments.

File: docs/source/tutorial/neighborloader.rst (hunk 1 above)
Comment (LukeLIN-web) [reply docs/source/tutorial/neighborloader.rst:124]: Do you mean we need to explain why we need a layer-wise way?

Comment (LukeLIN-web) [reply docs/source/tutorial/neighborloader.rst:137]: > We would need to go a bit more into depth here if we wanna take about layer-wise computation of node embeddings for inference.

It seems out of scope. I would like to remove this part.

**Review by rusty1s — APPROVED** [review]

**Pull request merged**